lm-feedCore-iOS/lm-feedCore-iOS/Source/Classes/Screens/SearchPost/LMFeedSearchPostViewModel.swift

```diff
@@ -0,0 +1,504 @@
+//
+//  LMFeedSearchPostViewModel.swift
+//  LikeMindsFeedCore
+//
+//  Created by Devansh Mohata on 29/06/24.
+//
+
+import LikeMindsFeed
+import LikeMindsFeedUI
+
+public protocol LMFeedSearchPostViewModelProtocol: LMBaseViewControllerProtocol {
+    func showLoader(isShow: Bool)
+    func showTableFooter(isShow: Bool)
+    func updatePostList(with post: [LMFeedPostContentModel])
+    func removePost(postID: String)
+    func updatePost(post: LMFeedPostContentModel)
+    func toggleEmptyView(isShow: Bool)
+    func removePreviousResults()
+    
+    func navigateToDeleteScreen(for postID: String)
+    func navigateToReportScreen(for postID: String, creatorUUID: String)
+    func navigateToPollResultScreen(with pollID: String, optionList: [LMFeedPollDataModel.Option], selectedOption: String?)
+    func navigateToAddOptionPoll(with postID: String, pollID: String, options: [String])
+}
+
+public class LMFeedSearchPostViewModel {
+    var timer: Timer?
+    unowned var delegate: LMFeedSearchPostViewModelProtocol
+    var postList: [LMFeedPostDataModel]
+    var page: Int
+    var searchString: String
+    let pageSize: Int
+    let searchType: String
+    var isFetchingResults: Bool
+    var isLastPage: Bool
+    
+    init(delegate: LMFeedSearchPostViewModelProtocol) {
+        self.timer = nil
+        self.delegate = delegate
+        self.postList = []
+        self.page = 1
+        self.pageSize = 10
+        self.searchString = ""
+        self.searchType = "text"
+        self.isFetchingResults = false
+        self.isLastPage = false
+    }
+    
+    public static func createModule() throws -> LMFeedSearchPostScreen {
+        guard LMFeedCore.isInitialized else { throw LMFeedError.feedNotInitialized }
+        
+        let viewController = Components.shared.searchPostScreen.init()
+        let viewModel = LMFeedSearchPostViewModel.init(delegate: viewController)
+        
+        viewController.viewModel = viewModel
+        
+        return viewController
+    }
+    
+    public func searchPosts(with search: String) {
+        timer?.invalidate()
+        
+        guard !search.isEmpty else {
```

Comment:
Suppose I search for a term and then erased the search text, will it show the EmptyView in that case? Even if the search call fails?

If the search text is empty, can we check if we have some results in the delegate. If we do we can replace it with an empty view and return from the function without making a call to the backend.